go/engine/track_token_test.go

```diff
@@ -320,3 +325,126 @@ func TestTrackRemoteThenLocalTemp(t *testing.T) {
 
 	assertTracking(tc, username)
 }
+
+func TestTrackFailTempRecover(t *testing.T) {
+	tc := SetupEngineTest(t, "track")
+	defer tc.Cleanup()
+	fu := CreateAndSignupFakeUser(tc, "track")
+
+	fakeClock := clockwork.NewFakeClock()
+	tc.G.Clock = fakeClock
+	// to pick up the new clock...
+	tc.G.ResetLoginState()
+
+	flakeyAPI := flakeyRooterAPI{orig: tc.G.XAPI, flakeOut: false, G: tc.G}
+	tc.G.XAPI = &flakeyAPI
+
+	idUI := &FakeIdentifyUI{}
+	username := "t_tracy"
+
+	arg := &keybase1.Identify2Arg{
+		UserAssertion: username,
+		NeedProofSet:  true,
+	}
+	ctx := &Context{
+		LogUI:      tc.G.UI.GetLogUI(),
+		IdentifyUI: idUI,
+		SecretUI:   fu.NewSecretUI(),
+	}
+
+	// Identify tracy; all proofs should work
+	eng := NewResolveThenIdentify2(tc.G, arg)
+	if err := RunEngine(eng, ctx); err != nil {
+		t.Fatal(err)
+	}
+	targ := TrackTokenArg{
+		Token:   idUI.Token,
+		Options: keybase1.TrackOptions{BypassConfirm: true, LocalOnly: true},
+	}
+
+	// Track tracy
+	teng := NewTrackToken(&targ, tc.G)
+	if err := RunEngine(teng, ctx); err != nil {
+		t.Fatal(err)
+	}
+
+	defer runUntrack(tc.G, fu, username)
+
+	// Now make her Rooter proof fail with a 429
+	flakeyAPI.flakeOut = true
+	idUI = &FakeIdentifyUI{}
+	ctx.IdentifyUI = idUI
+
+	// Advance so that our previous cached success is out of
+	// cache
+	fakeClock.Advance(tc.G.Env.GetProofCacheLongDur() + time.Minute)
+
+	eng = NewResolveThenIdentify2(tc.G, arg)
+	eng.testArgs = &Identify2WithUIDTestArgs{noCache: true}
+
+	// Should  get an error
+	if err := RunEngine(eng, ctx); err == nil {
+		t.Fatal("Expected identify error")
+	}
+
+	result, found := idUI.ProofResults["rooter"]
+	if !found {
+		t.Fatal("Failed to find a rooter proof")
+	}
+	pe := libkb.ImportProofError(result.ProofResult)
+	if pe == nil {
+		t.Fatal("expected a Rooter error result")
+	}
+
+	t.Logf("Rooter proof result, error: %v, -- %v", result, pe)
+	if result.Diff != nil {
+		t.Logf("Rooter proof result diff: %v", result.Diff)
+	}
+	// This is like the UI saying to store the local track
+	targ.Options.ExpiringLocal = true
+
+	targ.Token = idUI.Token
+	// Track tracy
+	teng = NewTrackToken(&targ, tc.G)
+	if err := RunEngine(teng, ctx); err != nil {
+		t.Fatal(err)
+	}
+
+	// Now make her Rooter proof work again
+	flakeyAPI.flakeOut = false
+
+	// Identify should work because of the original, permanent track
+	eng = NewResolveThenIdentify2(tc.G, arg)
+	eng.testArgs = &Identify2WithUIDTestArgs{noCache: true}
+	var err error
+	// Should not get an error
+	if err = RunEngine(eng, ctx); err != nil {
+		t.Logf("Identify failure: %v", err)
+		t.Fatal("Expected to pass identify")
+	}
+
+	// There shouldn't be a Diff in the result, but if there is, make sure
+	// it isn't due to temporary tracking
+	result, found = idUI.ProofResults["rooter"]
+	t.Logf("Rooter proof result, found for rooter: %v, %v", result, found)
+	if found {
+		if result.Diff != nil {
```

Comment:
When do you see a diff here?  Seems like you shouldn't be judging from the code, you do?....